File: test-parse-amount.js
// Test parsing of totalAmount from backend response
const testResponses = [
    { totalAmount: "1500.00" },  // String from test
    { totalAmount: 1500.00 },   // Number
    { totalAmount: "1000.00" },  // String
    { amount: 2000 },           // Using amount field
    {}
];

testResponses.forEach((response, index) => {
    console.log(`\nTest ${index + 1}:`, JSON.stringify(response));
    
    let parsedAmount;
    if (response.amount != null) {
        parsedAmount = typeof response.amount === 'string' 
            ? parseFloat(response.amount) 
            : Number(response.amount);
    } else if (response.totalAmount != null) {
        parsedAmount = typeof response.totalAmount === 'string' 
            ? parseFloat(response.totalAmount) 
            : Number(response.totalAmount);
    } else {
        parsedAmount = 0.0;
    }
    
    console.log(`Parsed amount: ${parsedAmount} (Type: ${typeof parsedAmount})`);
});

File: test_parse_amount.dart
import 'dart:convert';

void main() {
  final testResponses = [
    '{"totalAmount": "1500.00"}',
    '{"totalAmount": 1500.00}',
    '{"totalAmount": "1000.00"}',
    '{"amount": 2000}',
    '{}',
  ];

  testResponses.asMap().forEach((index, responseJson) {
    print('\nTest ${index + 1}: $responseJson');
    try {
      final Map<String, dynamic> json = jsonDecode(responseJson);

      double? parsedAmount;

      if (json['amount'] != null) {
        parsedAmount = (json['amount'] is String)
            ? double.tryParse(json['amount'])
            : json['amount']?.toDouble();
      } else if (json['totalAmount'] != null) {
        parsedAmount = (json['totalAmount'] is String)
            ? double.tryParse(json['totalAmount'])
            : json['totalAmount']?.toDouble();
      } else {
        parsedAmount = 0.0;
      }

      print(
        'Parsed amount: $parsedAmount (Type: ${parsedAmount?.runtimeType})',
      );
    } catch (e) {
      print('Error parsing: $e');
    }
  });
}
